src/parse/arg_matcher.rs

```diff
@@ -136,8 +136,14 @@ impl ArgMatcher {
         self.insert(arg);
     }
 
+    pub fn set_env_set(&mut self, arg: &Id) {
+        let ma = self.entry(arg).or_insert(MatchedArg::new());
+        ma.env_set = true;
+    }
+
```

Comment:
Why do you need this helper? It's a two line function and you're using it in only one place.

FYI: if a function is not supposed to be exported to user (it's not a part of public API), you should use `pub(crate)` but not `pub`.

Reply:
> Why do you need this helper?

I liked the encapsulation it provided. Otherwise, I had to bring `MatchedArg` into `parser.rs`. Happy to remove it if you prefer.

Reply:
The name here is not readable. Once you rename the function and the field, you should probably rename this.